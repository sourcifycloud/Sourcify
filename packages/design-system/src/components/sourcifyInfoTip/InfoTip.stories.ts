import type { StoryFn } from '@storybook/vue';
import sourcifyInfoTip from './InfoTip.vue';

export default {
	title: 'Atoms/InfoTip',
	component: sourcifyInfoTip,
};

const Template: StoryFn = (args, { argTypes }) => ({
	props: Object.keys(argTypes),
	components: {
		sourcifyInfoTip,
	},
	template:
		'<sourcify-info-tip v-bind="$props">Need help doing something? <a href="/docs" target="_blank">Open docs</a></sourcify-info-tip>',
});

export const Note = Template.bind({});

export const Tooltip = Template.bind({});
Tooltip.args = {
	type: 'tooltip',
	tooltipPlacement: 'right',
};
