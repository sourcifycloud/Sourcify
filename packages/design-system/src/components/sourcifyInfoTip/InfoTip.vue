<template>
	<div
		:class="{
			'sourcify-info-tip': true,
			[$style[theme]]: true,
			[$style[type]]: true,
			[$style.bold]: bold,
		}"
	>
		<sourcify-tooltip
			v-if="type === 'tooltip'"
			:placement="tooltipPlacement"
			:popper-class="$style.tooltipPopper"
			:disabled="type !== 'tooltip'"
		>
			<span :class="$style.iconText">
				<sourcify-icon :icon="theme.startsWith('info') ? 'info-circle' : 'exclamation-triangle'" />
			</span>
			<template #content>
				<span>
					<slot />
				</span>
			</template>
		</sourcify-tooltip>
		<span :class="$style.iconText" v-else>
			<sourcify-icon :icon="theme.startsWith('info') ? 'info-circle' : 'exclamation-triangle'" />
			<span>
				<slot />
			</span>
		</span>
	</div>
</template>

<script lang="ts">
import sourcifyIcon from '../sourcifyIcon';
import sourcifyTooltip from '../sourcifyTooltip';

import { defineComponent } from 'vue';

export default defineComponent({
	name: 'sourcify-info-tip',
	components: {
		sourcifyIcon,
		sourcifyTooltip,
	},
	props: {
		theme: {
			type: String,
			default: 'info',
			validator: (value: string): boolean =>
				['info', 'info-light', 'warning', 'danger'].includes(value),
		},
		type: {
			type: String,
			default: 'note',
			validator: (value: string): boolean => ['note', 'tooltip'].includes(value),
		},
		bold: {
			type: Boolean,
			default: true,
		},
		tooltipPlacement: {
			type: String,
			default: 'top',
		},
	},
});
</script>

<style lang="scss" module>
.base {
	font-size: var(--font-size-2xs);
	line-height: var(--font-size-s);
	word-break: normal;
	display: flex;
	align-items: center;

	svg {
		font-size: var(--font-size-s);
	}
}

.bold {
	font-weight: var(--font-weight-bold);
}

.note {
	composes: base;

	svg {
		margin-right: var(--spacing-4xs);
	}
}

.tooltip {
	composes: base;
	display: inline-flex;
}

.iconText {
	display: inline-flex;
	align-items: flex-start;
}

.info-light {
	color: var(--color-foreground-dark);
}

.info {
	color: var(--color-text-light);
}

.warning {
	color: var(--color-warning);
}

.danger {
	color: var(--color-danger);
}
</style>
