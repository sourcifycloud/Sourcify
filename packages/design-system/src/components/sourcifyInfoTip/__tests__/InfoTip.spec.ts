import { render } from '@testing-library/vue';
import sourcifyInfoTip from '../InfoTip.vue';

const slots = {
	default: ['Need help doing something?', '<a href="/docs" target="_blank">Open docs</a>'],
};
const stubs = ['sourcify-tooltip'];

describe('sourcifyInfoTip', () => {
	it('should render correctly as note', () => {
		const wrapper = render(sourcifyInfoTip, {
			slots,
			stubs,
		});
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should render correctly as tooltip', () => {
		const wrapper = render(sourcifyInfoTip, {
			slots,
			stubs,
			props: {
				type: 'tooltip',
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});
});
