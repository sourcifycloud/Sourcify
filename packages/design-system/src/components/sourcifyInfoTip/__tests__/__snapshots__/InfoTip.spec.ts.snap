// Vitest Snapshot v1

exports[`sourcifyInfoTip > should render correctly as note 1`] = `"<div class=\\"sourcify-info-tip info note bold\\"><span class=\\"iconText\\"><span class=\\"sourcify-icon sourcify-text compact size-medium regular\\"></span><span>Need help doing something?<a href=\\"/docs\\" target=\\"_blank\\">Open docs</a></span></span></div>"`;

exports[`sourcifyInfoTip > should render correctly as tooltip 1`] = `
"<div class=\\"sourcify-info-tip info tooltip bold\\">
  <sourcify-tooltip-stub justifybuttons=\\"flex-end\\" buttons=\\"\\" placement=\\"top\\" popper-class=\\"tooltipPopper\\"><span class=\\"iconText\\"><span class=\\"sourcify-icon sourcify-text compact size-medium regular\\"></span></span></sourcify-tooltip-stub>
</div>"
`;
